File: 0206-reverse-linked-list/0206-reverse-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function reverseList(head: ListNode | null): ListNode | null {
    let root = head;
    let lst = [];
    while (root !== null) {
        lst.push(root.val);
        root = root.next;
    }
    lst.reverse();
    root = head;
    lst.forEach ((val) => {
        root.val = val;
        root = root.next;
    });
    return head;
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* dum = head;
        ListNode* prev = nullptr;

        while (dum != nullptr) {
            ListNode* temp = dum->next;
            dum->next = prev;
            prev = dum;
            dum = temp;
        }

        return prev;
    }
};